Configtool refactoring

phord opened the pull request:
I was refactoring code in configtool to separate "model" and "view" code a little bit when I noticed your changes last night.  I rebased my changes in-flight over onto the i2c-rebuild tree, cleaned them up and tested them a bit, and added some test code to show my intended direction.  I drop it here now for your consideration.

It looks messier than it is because a lot of code moved into new files.  I expect I might move more soon.

Traumflug commented:
Excellent! Especially that you moved these commits to a branch already.

To reduce the need for rebases I took the plunge and moved all the already reviwed I2C code to *experimental* and also the first three of this pull request. 37 commits in total.

I didn't continue, because the now first commit on *configtool-refactoring* breaks Configtool. Just launching it gives a stacktrace:

    TypeError: loadConfigFile() takes exactly 3 arguments (2 given)

Usually I fix simple things, but in this case I'm not sure what your intention is. Code for the same task in *board.py* and *printer.py* differs.

*--show-all* looks a bit like a duplicate of *-vv*. Simply drop *--show-all*?

I took the liberty to force-push *configtool-refactoring* with my nitpickings already applied and rebased to the new *experimental*.

Other than that: great work.

phord commented:
I rebased and squashed done fixups before pushing this, and it looks like I dropped a couple of lines. I'll follow up later today.

I consider -vv and --show-all to serve very different purposes. -vv is for debugging, but --show-all is a script support feature. Also show-all is the first of several related commands.

phord commented:
Also, I did intend to make the python2 dependency go away in the cmdline version (since it would also not depend on wx), but I didn't follow through on testing this, obviously.  Some more py3 work needs to be done before it can be effective there, and the warning then needs to be moved so it is encountered before the first wx import somewhere.  Happy to put this off for some other day.  :zzz:
